R/download_ts.R

```diff
@@ -1,47 +1,56 @@
-#'@title download timeseries data to local file destination
-#'@description download a timeseries file to a user-specified (or temp file)
-#'  location
-#'@param site a valid mda.streams site (see \link{get_sites})
-#'@param var_src a valid variable name for timeseries data (see
-#'  \code{get_var_codes(out='var_src', type='ts')})
-#'@param folder string for a folder location
-#'@param ... additional arguments passed to
-#'  \code{\link[sbtools]{session_check_reauth}}
-#'  
-#'@return file handle for downloaded file
-#'@author Corinna Gries, Jordan S Read, Luke A Winslow
-#'@examples
-#'\dontrun{
-#'download_ts(site = 'nwis_06893300', var_src = 'doobs_nwis')
-#'}
-#'@import sbtools
-#'@import tools
-#'@export
-download_ts=function(site, var_src, folder = tempdir(), ...){
+#' @title download timeseries data to local file destination
+#' @description download a timeseries file to a user-specified (or temp file) 
+#'   location
+#' @param site a valid mda.streams site (see \link{get_sites})
+#' @param var_src a valid variable name for timeseries data (see 
+#'   \code{get_var_codes(out='var_src', type='ts')})
+#' @param folder string for a folder location
+#' @param on_exists character indicating what to do if the folder already 
+#'   contains a file with the intended download name
+#' @param ... additional arguments passed to 
+#'   \code{\link[sbtools]{session_check_reauth}}
+#'   
+#' @return file handle for downloaded file
+#' @author Corinna Gries, Jordan S Read, Luke A Winslow
+#' @examples
+#' \dontrun{
+#' download_ts(site = 'nwis_06893300', var_src = 'doobs_nwis')
+#' }
+#' @import sbtools
+#' @import tools
+#' @export
+download_ts=function(site, var_src, folder = tempdir(), on_exists=c("stop","skip","replace"), ...){
 
-  ts_variable <- paste0(pkg.env$ts_prefix, var_src)
-  scheme = get_scheme()
+  # check inputs
+  on_exists <- match.arg(on_exists)
```

Comment:
maybe the default is `NULL`, and you do 

`on_exists <- match.arg(on_exists,c("stop","skip","replace"))` here?